### PR TITLE
Add EIP: Token Revenue Sharing

### PR DESCRIPTION
**Pull request by quyphandang**
When opening a pull request to submit a new EIP, please use the suggested template: https://github.com/ethereum/EIPs/blob/master/eip-template.md

We have a GitHub bot that automatically merges some PRs. It will merge yours immediately if certain criteria are met:

 - The PR edits only existing draft PRs.
 - The build passes.
 - Your GitHub username or email address is listed in the 'author' header of all affected PRs, inside <triangular brackets>.
 - If matching on email address, the email address is the one publicly listed on your GitHub profile.

### CODE REVIEW

File: EIPS/eip-7254.md [hunk 1]
```diff
@@ -0,0 +1,174 @@
...
+
+
+
+#### balanceOf
+
+Returns the account balance of another account with address `account`.
+
+``` js
+function balanceOf(address account) public view returns (uint256)
+```
```
Comment (Pandapip1) [inline EIPS/eip-7254.md:68]: A lot of this is identical to ERC-20. I recommend extending it for compatibility with other EIPs.

File: Gemfile.lock [hunk 2]
```diff
@@ -290,6 +290,7 @@ GEM
     zeitwerk (2.6.7)
 
 PLATFORMS
+  arm64-darwin-20
```
Comment (Pandapip1) [inline Gemfile.lock:293]:
```suggestion
```
Make this change in a separate PR.

File: assets/eip-7254/ERC7254.sol [hunk 3]
```diff
@@ -0,0 +1,458 @@
+// SPDX-License-Identifier: MIT
```
Comment (Pandapip1) [inline assets/eip-7254/ERC7254.sol:1]:
```suggestion
// SPDX-License-Identifier: CC0-1.0
```
Please use CC0 to avoid license proliferation.

File: assets/eip-7254/IERC7254.sol [hunk 4]
```diff
@@ -0,0 +1,129 @@
+ // SPDX-License-Identifier: MIT
```
Comment (Pandapip1) [inline assets/eip-7254/IERC7254.sol:1]:
```suggestion
 // SPDX-License-Identifier: CC0-1.0
```

File: assets/eip-7254/IERC7254Metadata.sol [hunk 5]
```diff
@@ -0,0 +1,27 @@
+// SPDX-License-Identifier: MIT
```
Comment (Pandapip1) [inline assets/eip-7254/IERC7254Metadata.sol:1]:
```suggestion
// SPDX-License-Identifier: CC0-1.0
```

File: EIPS/eip-7254.md [hunk 6]
```diff
@@ -0,0 +1,174 @@
...
+
+
+
+#### informationOf
+
+Returns the account information of another account with the address `account`, including: inReward, outReward and withdraw.
+
+``` js
+function informationOf(address account) public view returns (UserInformation memory)
```
Comment (Pandapip1) [inline EIPS/eip-7254.md:77]: `UserInformation` is not defined anywhere in this EIP.

File: EIPS/eip-7254.md [hunk 7]
```diff
@@ -0,0 +1,174 @@
...
+
+``` js
+function allowance(address owner, address spender) public view returns (uint256)
+```
+
+
+
+#### tokenReward
+
+Returns the token reward address of the token.
```
Comment (Pandapip1) [inline EIPS/eip-7254.md:123]: "Reward" has not been defined in a normative section, nor have the mechanics of the "reward" been specified.

Comment (Pandapip1) [inline EIPS/eip-7254.md:123]: Additionally, this limits the token to only providing a single token 'reward' - why not let the token be more extendable, and have the relevant functions return an array?

**Review by Pandapip1 — CHANGES_REQUESTED** [review]

File: EIPS/eip-7254.md (hunk 7 above)
Comment (quyphandang) [reply EIPS/eip-7254.md]: I have updated, but will limit the maxTokenReward because it will increase the user fee for each transaction.

File: EIPS/eip-7254.md (hunk 1 above)
Comment (quyphandang) [reply EIPS/eip-7254.md]: I have updated

**Comment by SamWilsn** [comment]
I am closing this pull request because we are in the process of separating EIPs and ERCs into distinct repositories. Unfortunately, as far as we are aware, GitHub does not provide any tools to ease this migration, so every pull request will need to be re-opened manually.

As this is a PR to create / modify an ERC, I will kindly ask you to redirect this to the new repository at [ethereum/ERCs](https://github.com/ethereum/ERCs). **We have prepared [a guide](https://github.com/ethereum/ERCs/issues/8) to help with the process.**

If there is relevant history here, please link to this PR from the new pull request.

On behalf of the EIP Editors, I apologize for this inconvenience.
